python/ray/data/dataset.py

```diff
@@ -353,8 +353,9 @@ def map_batches(
             fn: The function to apply to each record batch, or a class type
                 that can be instantiated to create such a callable. Callable classes are
                 only supported for the actor compute strategy.
-            batch_size: Request a specific batch size, or None to use entire
-                blocks as batches. Defaults to a system-chosen batch size.
+            batch_size: Request a specific batch size, or None to use entire blocks
+                as batches (blocks may contain different number of rows).
+                Defaults to 4096.
```

Comment:
Is this a random pick or do we have some use cases around this number?

Also I wonder if changing this default will have impact on performance tests?

Reply:
+1, we need to decide on perf vs stability by default, e.g. for many workloads (like CV training), 256 may be a better default.

Note that users are more likely to come to `.iter_batches()` with a batch size in mind than for non-inference `.map_batches()`, since the former will be tailored to the width of their data and the memory constraints of their model (which they're likely to have already thought about), while the latter is about making sure that their preprocessing UDF doesn't exhaust worker heap memory (which is more of a Datasets detail that they maybe haven't thought about yet).

Reply:
My vote would be starting with something at the scale of 256 or 128 by default, or we could match other libraries like Keras and use 32 as the default batch size: https://www.tensorflow.org/api_docs/python/tf/keras/Sequential#fit

Reply:
+1. How about 256 then? It seems like a good middle ground number.

Reply:
Yeah I can change it to 256 - would this apply to `map_batches` as well?

Reply:
No, just iter_batches(). map_batches() should optimize for performance since it's not connected to sgd.

Reply:
Yeah this change should just be for `.iter_batches()`!